server/src/main/java/org/apache/iotdb/db/engine/migration/TsFileMigrationLogger.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.engine.migration;
+
+import org.apache.iotdb.db.engine.fileSystem.SystemFileFactory;
+import org.apache.iotdb.db.engine.storagegroup.TsFileResource;
+import org.apache.iotdb.tsfile.utils.FilePathUtils;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.apache.iotdb.db.metadata.idtable.IDTable.config;
+import static org.apache.iotdb.db.metadata.idtable.IDTable.logger;
+
+/**
+ * To assure that migration of tsFiles is pesudo-atomic operator, TsFileMigrationLogger writes files
+ * to migratingFileDir when a migration task migrates a tsFile (and its resource/mod files) , then
+ * deletes it after the task has finished operation.
+ */
+public class TsFileMigrationLogger implements AutoCloseable {
+
+  // taskId -> MigratingFileLog
+  ConcurrentHashMap<Long, FileOutputStream> logOutputMap = new ConcurrentHashMap<>();
```

Comment:
1. Please use map interface
2. It's better for each migration task to control it's own resources, like FileOutputStream here. It‘s unsuitable for MigrationTask class to only stores some information, but does‘t take any responsibilities to actions. So, maybe you should move these responsibilities into MigrationTask itself and delete this class.

Reply:
1. Fixed.
2. Moved TsFileMigrationLogger responsibilities into MigrationTask.